src/runtime/slots/mp_length.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Python.Runtime.Slots
+{
+    internal static class mp_length_slot
+    {
+        /// <summary>
+        /// Finds a get_Count (Count property) that is private and
+        /// is either ICollection or ICollection&lt;T&gt; explicit
+        /// implementation
+        /// </summary>
+        static MethodInfo GetCountGetter(Type t)
```

Comment:
Instead of doing this (which looks rather obscure), you could simply do:

```csharp
var genericCollectionInterface = clrType.GetInterfaces()
  // this should return typeof(ICollection<int>)
  .FirstOrDefault(x => x.IsGenericType && x.GetGenericTypeDefinition() == typeof(ICollection<>));
if (genericCollectionInterface == null) { ...error... }

// this won't be null, as `ICollection<int>` has a "public" property `Count`
var countProperty = genericCollectionInterface.GetProperty(nameof(ICollection<int>.Count));
return countProperty.GetValue(co.inst);
...
```

Reply:
I tried doing something similar to that, but it said that the target type is not the same as the property type when I was running it. I'll revisit to make sure I didn't miss something.

Reply:
It works for arrays in interactive:
```csharp
> typeof(int[]).GetInterfaces()
  .FirstOrDefault(x => x.IsGenericType && x.GetGenericTypeDefinition() == typeof(ICollection<>))
  .GetProperty(nameof(ICollection<int>.Count))
  .GetValue(new int[42])
42
```